ldclient_test.go

```diff
@@ -30,3 +35,50 @@ func TestOfflineModeAlwaysReturnsDefaultValue(t *testing.T) {
 		t.Errorf("Offline mode should return default value, but doesn't")
 	}
 }
+
+type evaluateTestData struct {
+	FeatureKey    string      `json:"featureKey"`
+	DefaultValue  string      `json:"defaultValue"`
+	ExpectedValue string      `json:"expectedValue"`
+	ExpectError   bool        `json:"expectError"`
+	User          User        `json:"user"`
+	FeatureFlag   FeatureFlag `json:"featureFlag"`
+}
+
+func TestEvaluate(t *testing.T) {
+	var container []evaluateTestData
+	file, err := ioutil.ReadFile("./test_data.json")
+	if err != nil {
+		t.Errorf("Error loading test_data.json file: %v\n", err)
+		return
+	}
+	err = json.Unmarshal(file, &container)
+	if err != nil {
+		t.Errorf("Error unmarshalling test_data.json file: %v\n", err)
+		return
+	}
+
+	for _, td := range container {
+		json, _ := json.MarshalIndent(td, "", "  ")
+		t.Logf("Test data: %s", string(json))
+		client, _ = MakeCustomClient("api_key", config, 0)
```

Comment:
I can go either way on loading all the flags into the same client instance, or keeping them separate here.